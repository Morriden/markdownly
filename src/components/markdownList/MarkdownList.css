.MarkdownList {
  ul {
    display: inline-flex;
   
    width: 50vh;
    overflow: scroll;
  }
  li {
    height: 1rem;
    border: 2px solid black;
    border-radius: 15px;
    padding: 10px;
    text-align: center;
    margin: 1rem;
  }
}