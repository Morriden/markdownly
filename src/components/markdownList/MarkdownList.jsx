import React from 'react';
import styles from './MarkdownList.css';
import { getMarkdownList } from '../../selectors/markdownSelectors';
import { useSelector } from '../../hooks/appContext';
import { Link } from 'react-router-dom';

const MarkdownList = () => {
  const markdownList = useSelector(getMarkdownList);
  const markdownListElements = markdownList.map((markdown) => (
    <Link to={`/${markdown.id}`} key={markdown.id} >
      <li key={markdown.id}>
        {markdown.title}
      </li>
    </Link>
  ));

  return (
    <div className={styles.MarkdownList}>
      <ul>
        {markdownListElements}
      </ul>
    </div>
  );
};

export default MarkdownList;
