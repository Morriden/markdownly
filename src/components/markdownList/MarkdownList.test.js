import React from 'react';
import { shallow } from 'enzyme';
import MarkdownList from './MarkdownList';
import AppProvider from '../../provider/AppProvider';


describe('MarkdownList component', () => {
  it('renders MarkdownList', () => {
    const wrapper = shallow(<AppProvider><MarkdownList /></AppProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
