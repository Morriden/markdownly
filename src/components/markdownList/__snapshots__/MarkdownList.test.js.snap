// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownList component renders MarkdownList 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "state": Object {
        "markdownList": Array [
          Object {
            "body": "Look at that markdown file.",
            "id": 1234,
            "title": "whatever",
          },
          Object {
            "body": "Look at this new markdown file.",
            "id": 5678,
            "title": "new tab",
          },
        ],
      },
    }
  }
>
  <MarkdownList />
</ContextProvider>
`;
